x-pack/plugins/threat_intelligence/public/modules/query_bar/hooks/use_filter_in_out.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback } from 'react';
+import { Filter } from '@kbn/es-query';
+import { useIndicatorsFiltersContext } from '../../indicators/hooks/use_indicators_filters_context';
+import { fieldAndValueValid, getIndicatorFieldAndValue } from '../../indicators/lib/field_value';
+import { FilterIn, FilterOut, updateFiltersArray } from '../lib/filter';
+import { Indicator } from '../../../../common/types/indicator';
+
+export interface UseFilterInParam {
+  /**
+   * Indicator used to retrieve the field and value then use to update the filters
+   */
+  indicator: Indicator | string;
+  /**
+   * Value used to filter in /out in the KQL bar.
+   */
+  field: string;
+  /**
+   * To filter in or out.
+   */
+  filterType: typeof FilterIn | typeof FilterOut;
+}
+
+export interface UseFilterInValue {
+  /**
+   * Filter function to run on click event.
+   */
+  filterFn: (() => void) | undefined;
+}
+
+/**
+ * Custom hook that uses an indicator, a field and a type (FilterIn or FilterOut) and returns the filter function.
+ *
+ */
+export const useFilterInOut = ({
+  indicator,
+  field,
+  filterType,
+}: UseFilterInParam): UseFilterInValue => {
+  const { filterManager } = useIndicatorsFiltersContext();
```

Comment:
it should be possible to simply do `useFilters` here, after the filters context refactor is merged:)